numpy/core/src/multiarray/number.c

```diff
@@ -550,6 +551,54 @@ array_power(PyArrayObject *a1, PyObject *o2, PyObject *modulo)
     return value;
 }
 
+static PyObject *
+array_positive(PyArrayObject *m1)
+{
+    /*
+     * For backwards compatibility, where + just implied a copy,
+     * we cannot just call n_ops.positive.  Instead, we do the following
+     * 1. Try n_ops.positive
+     * 2. If we get an exception, check whether __array_ufunc__ is
+     *    overridden; if so, we live in the future and we allow the
+     *    TypeError to be passed on.
+     * 3. If not, give a deprecation warning and return a copy.
+     */
+    PyObject *value;
+    PyObject *mod;
+    if (mod) {
+        Py_DECREF(mod);
+    }
```

Comment:
This looks wrong. What is mod?

Reply:
Oops, leftover from debugging!! (I had the problem that `has_not_default...` doesn't work when an error is set)